### DISCUSSION TITLE
Expenses Installation

### DISCUSSION THREAD

**Discussion by DustyArmstrong** [kimai/kimai#2913] [On-Premise: Installation & Updates] [answered]
Hi there, 

I'm looking to implement the expenses plugin, but I have a couple of questions. 

1) How is this licensed? Is it registered to an email or is it simply provided as the zip file? 

2) I'm running Kimai as a docker container `kimai/kimai2:fpm`. The installation instructions state that the expenses plugin should be placed in `/var/plugins/`. Given how docker works I imagine I should migrate this path to a docker volume for persistence? It also looks like that path needs to be created, is this correct?

Thanks.

**Comment by kevinpapst** [comment, marked as answer]
I am not a docker user and cannot help with that issue. 
But I believe it is a good idea anyways to have the `var/`directory (inside kimai! not `/var/`) directory being a mount, as it includes stuff like generated invoices, invoice and export templates and so on.

You get a ZIP file and one invitation to the GIT repository as stated in the docs: 
https://www.kimai.org/documentation/buy-a-plugin.html

There is no license key, remote check or any other crappy idea of DRM. I trust you, the same way as you trust me.

**Comment by DustyArmstrong** [comment]
Thanks for the info, that should be fine for me. 

Docker volumes are technically still "inside" the container, they are just mapped onto the host file system. To the container, they exist exactly as if they were local: it's because docker containers are "fresh" every time you pull an image, you would lose all user data unless you persist with a mapped volume. Sorry if I'm explaining something you already understand and I've misunderstood.

**Reply by kevinpapst** [reply]
I kinda know more about Docker than I admit 😁 but still after all these years I cannot become friend with it ... believe me, I tried for month. It's just not mine and I do believe that Kimai is enough for one person. 

Just to clarify, because your initial post is misleading: Plugins are installed relative to your Kimai installation at [var/plugins/](https://github.com/kevinpapst/kimai2/tree/master/var/plugins) and not in the absolute location `/var/plugins/`.

**Reply by DustyArmstrong** [reply]
Thanks! I love docker, it has worked very well for me so I use Kimai as a docker container. 

Yes I also went to the Kimai docker repo (maintained by another user) and the path will be `/opt/kimai/var/plugins/[name of plugin]`
